tests/finance/test_slippage.py

```diff
@@ -24,75 +24,145 @@
 
 from nose_parameterized import parameterized
 
+import numpy as np
 import pandas as pd
+from testfixtures import TempDirectory
 
 from zipline.finance.slippage import VolumeShareSlippage
+from zipline.finance.trading import TradingEnvironment, SimulationParameters
 
-from zipline.protocol import Event, DATASOURCE_TYPE
+from zipline.protocol import DATASOURCE_TYPE
 from zipline.finance.blotter import Order
 
+from zipline.data.minute_writer import MinuteBarWriterFromDataFrames
+from zipline.data.data_portal import DataPortal
+
 
 class SlippageTestCase(TestCase):
 
-    def test_volume_share_slippage(self):
-        event = Event(
-            {'volume': 200,
-             'type': 4,
-             'price': 3.0,
-             'datetime': datetime.datetime(
-                 2006, 1, 5, 14, 31, tzinfo=pytz.utc),
-             'high': 3.15,
-             'low': 2.85,
-             'sid': 133,
-             'source_id': 'test_source',
-             'close': 3.0,
-             'dt':
-             datetime.datetime(2006, 1, 5, 14, 31, tzinfo=pytz.utc),
-             'open': 3.0}
+    @classmethod
+    def setUpClass(cls):
+        cls.tempdir = TempDirectory()
+        cls.env = TradingEnvironment()
+
+        cls.sim_params = SimulationParameters(
+            period_start=pd.Timestamp("2006-01-05 14:31", tz="utc"),
+            period_end=pd.Timestamp("2006-01-05 14:36", tz="utc"),
+            capital_base=1.0e5,
+            data_frequency="minute",
+            emission_rate='daily',
+            env=cls.env
         )
 
-        slippage_model = VolumeShareSlippage()
+        cls.sids = [133]
 
-        open_orders = [
-            Order(dt=datetime.datetime(2006, 1, 5, 14, 30, tzinfo=pytz.utc),
-                  amount=100,
-                  filled=0,
-                  sid=133)
-        ]
+        cls.minutes = pd.DatetimeIndex(
+            start=pd.Timestamp("2006-01-05 14:31", tz="utc"),
+            end=pd.Timestamp("2006-01-05 14:35", tz="utc"),
+            freq="1min"
+        )
 
-        orders_txns = list(slippage_model.simulate(
-            event,
-            open_orders
-        ))
+        assets = {
+            133: pd.DataFrame({
+                "open": np.array([3.0, 3.0, 3.5, 4.0, 3.5]) * 1000,
+                "high": np.array([3.15, 3.15, 3.15, 3.15, 3.15]) * 1000,
+                "low": np.array([2.85, 2.85, 2.85, 2.85, 2.85]) * 1000,
+                "close": np.array([3.0, 3.5, 4.0, 3.5, 3.0]) * 1000,
+                "volume": [2000, 2000, 2000, 2000, 2000],
+                "minute": cls.minutes
+            })
+        }
 
-        self.assertEquals(len(orders_txns), 1)
-        _, txn = orders_txns[0]
+        MinuteBarWriterFromDataFrames().write(cls.tempdir.path, assets)
```

Comment:
Could be convenient and probably much faster for tests to have a DataPortal that reads in-memory data, versus a BcolzDataPortal that requires bcolz files on disk.  If the bcolz reader only needs a file stream, then maybe a StreamIO could work as a compatibility layer.